### PR TITLE
feat: waitlist API

### PR DESCRIPTION
**Pull request by NirmalHegde**
## Notion ticket link

[Ticket Name](https://uwblueprintexecs.notion.site/Waitlist-API-4d91be4d9031418d93f077dc6e412fc0)

## Implementation description
* Created Waitlist Routes
* Created WaitlistCamper Validator
* Created POST function for registering waitlisted campers

## Steps to test
1. Ensure API works as expected

## What should reviewers focus on?
* Ensure data is in correct format

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by annechung — COMMENTED** [review]
looks good! one thought I had though is that it might make things more organized to stick the createWaitlistedCamper method you made into the existing camperservice (I think a /camper/waitlist endpoint makes more sense than a waitlist/register endpoint)? just not sure if it's worth having a whole separate service for waitlist things. definitely open to discussion on this though, let me know what you think

**Comment by NirmalHegde** [comment]
> looks good! one thought I had though is that it might make things more organized to stick the createWaitlistedCamper method you made into the existing camperservice (I think a /camper/waitlist endpoint makes more sense than a waitlist/register endpoint)? just not sure if it's worth having a whole separate service for waitlist things. definitely open to discussion on this though, let me know what you think

I decided to make a waitlist api simply because I felt as though each one would have its own set of REST endpoints by the end. So it would be more organized in that aspect if it was divided, as they still have different class definitions and such. However, I can change it to what you mentioned for sure if you feel it would be better!

File: backend/typescript/services/implementations/camperService.ts
Comment (annechung) [inline backend/typescript/services/implementations/camperService.ts:200]:
```suggestion
            $push: { waitlist: newWaitlistedCamper.id },
```

**Review by annechung — COMMENTED** [review]
tested and just need the one fix for updating array of waitlistedcampers on the camp. should be good to merge after that!

**Review by annechung — COMMENTED** [review]
I think something went wrong while resolving merge conflicts D: getting errors on the backend

**Review by annechung — APPROVED** [review]
awesome work, feel free to merge this in! (after resolving merge conflicts D: )

**Pull request merged**
